### PR TITLE
some minor changes

### PR DESCRIPTION
**Pull request by ltroin**
also for the code refactoring, adding prettier, eslint

### CODE REVIEW

File: src/components/SampleScreen.js [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+  font-weight: ${(props) => props.theme.fontWeights.medium};
+`;
+
+export const SampleScreen = (screenName) => {
+  return (
+    <SafeArea>
+      <sampleView>
+        <Text>This screen ({screenName}) will be available soon!</Text>
+      </sampleView>
+    </SafeArea>
```
Comment (anhanh11001) [inline src/components/SampleScreen.js:24]: Question: What is this SafeArea for?

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,7686 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (anhanh11001) [inline yarn.lock:1]: Is this file yarn.lock needed? If it is not, please add it to `.gitignore`

File: App.js [hunk 3]
```diff
@@ -1,19 +1,44 @@
...
+import {
+  useFonts as useOstt,
+  OldStandardTT_400Regular,
+  OldStandardTT_400Regular_Italic,
+  OldStandardTT_700Bold,
+} from "@expo-google-fonts/old-standard-tt";
 
+import { theme } from "./src/infrastructure/theme";
+import HomeMadeTabs from "./src/infrastructure/navigations/Tabs";
+import HomeScreen from "./src/screens/home/HomeScreen";
```
Comment (anhanh11001) [inline App.js:16]: IMHO, I don't think it is we need a `/infrastructure` folder. We can keep `/navigations` and `/theme` inside `src`

File: package.json [hunk 4]
```diff
@@ -1,35 +1,48 @@
...
+    "lint": "eslint . --ext .js"
   },
   "dependencies": {
-    "@react-native-community/masked-view": "^0.1.11",
+    "@babel/preset-env": "^7.1.6",
+    "@react-native-async-storage/async-storage": "~1.15.0",
     "@react-navigation/bottom-tabs": "^6.2.0",
-    "@react-navigation/material-bottom-tabs": "^6.1.1",
     "@react-navigation/native": "^6.0.8",
-    "@react-navigation/native-stack": "^6.5.0",
```
Comment (anhanh11001) [inline package.json:16]: I got this error:



```
Unable to resolve module @react-navigation/native-stack from /Users/duc.letran/Desktop/design-project-HomeMade/App.js: @react-navigation/native-stack could not be found within the project or in these directories:
  node_modules

If you are sure the module exists, try these steps:
 1. Clear watchman watches: watchman watch-del-all
 2. Delete node_modules and run yarn install
 3. Reset Metro's cache: yarn start --reset-cache
 4. Remove the cache: rm -rf /tmp/metro-*
  4 | import { ThemeProvider } from "styled-components/native";
  5 | import { NavigationContainer } from "@react-navigation/native";
> 6 | import { createNativeStackNavigator } from "@react-navigation/native-stack";
```

It is because the dependency is removed from `package.json` here. Why was this removed?

**Review by anhanh11001 — COMMENTED** [review]
Just a comment, your branch name now is too long. So suggest for next time, just keep it short. For example: `zihao/theming_and_project_structure`. More details can be added in the PR description.

**Comment by anhanh11001** [comment]
Have you tried compiling the app and testing it before making the Pull Request (PR) ?

File: src/components/SampleScreen.js (hunk 1 above)
Comment (ltroin) [reply src/components/SampleScreen.js:24]: render everything within the screen

Comment (ltroin) [reply src/components/SampleScreen.js:24]: e.g. does not render stuffs beyond the statusbar

Comment (anhanh11001) [reply src/components/SampleScreen.js:24]: I still don't understand it, so what wrong is we remove this `SafeArea`?

File: package.json (hunk 4 above)
Comment (ltroin) [reply package.json:16]: fixed

File: App.js (hunk 3 above)
Comment (ltroin) [reply App.js:16]: okay, I think it will work

File: yarn.lock (hunk 2 above)
Comment (ltroin) [reply yarn.lock:1]: as far as I know, it's needed, but thanks for reminding

File: src/components/SampleScreen.js (hunk 1 above)
Comment (ltroin) [reply src/components/SampleScreen.js:24]: mainly for different devices' compatibility, if removed, contents at the top of the screen may overlap with the statusbar. More info can be found here: https://docs.expo.dev/versions/latest/sdk/safe-area-context/
